uxarray/grid.py

```diff
@@ -0,0 +1,236 @@
+
+
+class Grid:
+    """The Uxarray Grid object class that describes an unstructured grid.
+
+    Examples
+    ----------
+    Open an exodus file with Uxarray Grid object
+
+    >>> mesh = ux.Grid("filename.g")
+
+    Save as ugrid file
+
+    >>> mesh.saveas("outfile.ug")
+    """
+
+    def __init__(self,
+                 *args,
+                 filepath=None,
+                 gridspec=None,
+                 vertices=None,
+                 concave=None,
+                 islatlon=None,
+                 mesh_filetype=None):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+
+        # TODO: Add or remove new Args/kwargs below as this develops further
+
+        Parameters
+        ----------
+        args : string, optional
+            - Input file name with extension or
+            - Vertex coordinates that form one face.
+        islatlon : bool, optional
+            Specify if the grid is lat/lon based:
+        concave: bool, optional
+            Specify if this grid has concave elements (internal checks for this are possible)
+        gridspec: bool, optional
+            Specifies gridspec
+        mesh_filetype: string, optional
+            Specify the mesh file type, eg. exo, ugrid, shp etc
+        kwargs : dict, optional
+            A dict initializing specific variables as stated above
+            - example: kwargs = {"concave" : True, "islatlon" : True"}
+
+        Raises
+        ------
+            RuntimeError: File not found
+        """
+        # TODO: fix when adding/exercising gridspec
+        # pull in optional arguments
+        self.filepath = filepath
+        self.gridspec = gridspec
+        self.vertices = vertices
+        self.islatlon = islatlon
+        self.concave = concave
+        self.mesh_filetype = mesh_filetype
+
+        # internal uxarray representation of mesh stored in internal object in_ds
+        self.in_ds = xr.Dataset()
+
+        # determine initialization type - string signifies a file, numpy array signifies a list of verts
+        in_type = type(args[0])
+
+        # check if initializing from verts:
+        try:
+            if not os.path.isfile(args[0]) and in_type is not np.ndarray:
+                raise FileNotFoundError("File not found: " + args[0])
+            elif in_type is np.ndarray:
+                self.vertices = args[0]
+                self.__from_vert__()
+        except ValueError as e:
+            # initialize from vertices
+            if in_type is np.ndarray:
+                self.vertices = args[0]
+                self.__from_vert__()
+            else:
+                raise RuntimeError(
+                    "Init called with args other than verts (ndarray) or filename (str)"
+                )
+        except TypeError as e:
+            raise RuntimeError("sting or np.ndarray supported as input")
```

Comment:
I'm still not 100% sure this is the best way to go about this. 

@erogluorhan @rajeeja Thoughts?

Reply:
@anissa111 I feel like it could be improved, too. Could you propose a different solution for comparison purposes?

Reply:
@erogluorhan I'll look in to this!

Reply:
@erogluorhan @rajeeja I have replaced this block with a more concise version. Let me know what your thoughts are! 

Just a note, I removed the gridspec initialization and replaced it with a todo for now since it doesn't appear to be implemented yet.

Reply:
Thanks @anissa111 your changes improve the readability of the code.

Reply:
Thanks for doing this!